audio/soundtouch/Portfile

```diff
@@ -26,7 +27,11 @@ checksums           rmd160  57c876f09ddac0cdca51411705b92064e9da6d95 \
 use_autoreconf      yes
 autoreconf.args     -fvi
 
-configure.args      --disable-silent-rules \
+# cc1plus: error: invalid option argument '-Ofast'
+compiler.blacklist-append {*gcc-[34].*} {clang < 500}
```

Comment:
Perhaps a better solution is also to [comment out this line](https://codeberg.org/soundtouch/soundtouch/src/commit/dd2252e9af3f2d6b749378173a4ae89551e06faf/configure.ac#L36) so that the compiler uses default CFLAGS from the MacPorts environment.

Reply:
Sure, that is also an option. Let me try it.

Reply:
Apparently it fails not only because of the flag:
```
--->  Building soundtouch
Executing:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2" && /usr/bin/make -j8 -w all 
make: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2'
/usr/bin/make  all-recursive
make[1]: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2'
Making all in include
make[2]: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/include'
/usr/bin/make  all-am
make[3]: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/include'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/include'
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/include'
Making all in source
make[2]: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/source'
Making all in SoundTouch
make[3]: Entering directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/source/SoundTouch'
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT AAFilter.lo -MD -MP -MF .deps/AAFilter.Tpo -c -o AAFilter.lo AAFilter.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT FIRFilter.lo -MD -MP -MF .deps/FIRFilter.Tpo -c -o FIRFilter.lo FIRFilter.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT FIFOSampleBuffer.lo -MD -MP -MF .deps/FIFOSampleBuffer.Tpo -c -o FIFOSampleBuffer.lo FIFOSampleBuffer.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT RateTransposer.lo -MD -MP -MF .deps/RateTransposer.Tpo -c -o RateTransposer.lo RateTransposer.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT SoundTouch.lo -MD -MP -MF .deps/SoundTouch.Tpo -c -o SoundTouch.lo SoundTouch.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT TDStretch.lo -MD -MP -MF .deps/TDStretch.Tpo -c -o TDStretch.lo TDStretch.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT cpu_detect_x86.lo -MD -MP -MF .deps/cpu_detect_x86.Tpo -c -o cpu_detect_x86.lo cpu_detect_x86.cpp
/bin/sh ../../libtool  --tag=CXX   --mode=compile /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include  -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include  -pipe -Os -arch ppc -MT BPMDetect.lo -MD -MP -MF .deps/BPMDetect.Tpo -c -o BPMDetect.lo BPMDetect.cpp
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT RateTransposer.lo -MD -MP -MF .deps/RateTransposer.Tpo -c RateTransposer.cpp  -fno-common -DPIC -o .libs/RateTransposer.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT AAFilter.lo -MD -MP -MF .deps/AAFilter.Tpo -c AAFilter.cpp  -fno-common -DPIC -o .libs/AAFilter.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT FIFOSampleBuffer.lo -MD -MP -MF .deps/FIFOSampleBuffer.Tpo -c FIFOSampleBuffer.cpp  -fno-common -DPIC -o .libs/FIFOSampleBuffer.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT SoundTouch.lo -MD -MP -MF .deps/SoundTouch.Tpo -c SoundTouch.cpp  -fno-common -DPIC -o .libs/SoundTouch.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT cpu_detect_x86.lo -MD -MP -MF .deps/cpu_detect_x86.Tpo -c cpu_detect_x86.cpp  -fno-common -DPIC -o .libs/cpu_detect_x86.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT FIRFilter.lo -MD -MP -MF .deps/FIRFilter.Tpo -c FIRFilter.cpp  -fno-common -DPIC -o .libs/FIRFilter.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT TDStretch.lo -MD -MP -MF .deps/TDStretch.Tpo -c TDStretch.cpp  -fno-common -DPIC -o .libs/TDStretch.o
libtool: compile:  /usr/bin/g++-4.2 -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -DSOUNDTOUCH_DISABLE_X86_OPTIMIZATIONS -I/opt/local/include -pipe -Os -arch ppc -MT BPMDetect.lo -MD -MP -MF .deps/BPMDetect.Tpo -c BPMDetect.cpp  -fno-common -DPIC -o .libs/BPMDetect.o
In file included from ../../include/FIFOSampleBuffer.h:42,
                 from AAFilter.h:41,
                 from RateTransposer.h:42,
                 from RateTransposer.cpp:38:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
In file included from AAFilter.h:41,
                 from RateTransposer.h:42,
                 from RateTransposer.cpp:38:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
In file included from RateTransposer.cpp:38:
RateTransposer.h:127: error: expected ';' before 'override'
RateTransposer.h:150: error: expected ';' before 'override'
RateTransposer.h:153: error: expected ';' before 'override'
RateTransposer.h:156: error: expected ';' before 'override'
RateTransposer.h:107: error: cannot declare field 'soundtouch::RateTransposer::inputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   because the following virtual functions are pure within 'soundtouch::FIFOSampleBuffer':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
RateTransposer.h:110: error: cannot declare field 'soundtouch::RateTransposer::midBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
RateTransposer.h:113: error: cannot declare field 'soundtouch::RateTransposer::outputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
In file included from RateTransposer.cpp:39:
InterpolateLinear.h:50: error: expected ';' before 'override'
InterpolateLinear.h:53: error: expected ';' before 'override'
InterpolateLinear.h:54: error: expected ';' before 'override'
InterpolateLinear.h:60: error: expected ';' before 'override'
InterpolateLinear.h:62: error: expected ';' before 'override'
In file included from RateTransposer.cpp:40:
InterpolateCubic.h:46: error: expected ';' before 'override'
InterpolateCubic.h:49: error: expected ';' before 'override'
InterpolateCubic.h:52: error: expected ';' before 'override'
InterpolateCubic.h:59: error: expected ';' before 'override'
In file included from RateTransposer.cpp:41:
InterpolateShannon.h:51: error: expected ';' before 'override'
InterpolateShannon.h:54: error: expected ';' before 'override'
InterpolateShannon.h:57: error: expected ';' before 'override'
InterpolateShannon.h:64: error: expected ';' before 'override'
RateTransposer.cpp: In constructor 'soundtouch::RateTransposer::RateTransposer()':
RateTransposer.cpp:64: warning: abstract virtual 'virtual void soundtouch::FIFOSamplePipe::clear()' called from constructor
RateTransposer.cpp: At global scope:
RateTransposer.cpp:68: error: definition of implicitly-declared 'virtual soundtouch::RateTransposer::~RateTransposer()'
RateTransposer.cpp:122: error: no 'void soundtouch::RateTransposer::putSamples(const soundtouch::SAMPLETYPE*, uint)' member function declared in class 'soundtouch::RateTransposer'
RateTransposer.cpp: In member function 'void soundtouch::RateTransposer::processSamples(const soundtouch::SAMPLETYPE*, uint)':
RateTransposer.cpp:139: error: no matching function for call to 'soundtouch::FIFOSampleBuffer::putSamples(const soundtouch::SAMPLETYPE*&, uint&)'
../../include/FIFOSampleBuffer.h:133: note: candidates are: virtual void soundtouch::FIFOSampleBuffer::putSamples(uint)
RateTransposer.cpp: At global scope:
RateTransposer.cpp:192: error: no 'void soundtouch::RateTransposer::clear()' member function declared in class 'soundtouch::RateTransposer'
RateTransposer.cpp:206: error: no 'int soundtouch::RateTransposer::isEmpty() const' member function declared in class 'soundtouch::RateTransposer'
RateTransposer.cpp: In static member function 'static soundtouch::TransposerBase* soundtouch::TransposerBase::newInstance()':
RateTransposer.cpp:305: error: cannot allocate an object of abstract type 'soundtouch::InterpolateCubic'
InterpolateCubic.h:42: note:   because the following virtual functions are pure within 'soundtouch::InterpolateCubic':
RateTransposer.h:64: note: 	virtual int soundtouch::TransposerBase::transposeMono(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:67: note: 	virtual int soundtouch::TransposerBase::transposeStereo(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:70: note: 	virtual int soundtouch::TransposerBase::transposeMulti(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:86: note: 	virtual void soundtouch::TransposerBase::resetRegisters()
RateTransposer.cpp:308: error: cannot allocate an object of abstract type 'soundtouch::InterpolateShannon'
InterpolateShannon.h:47: note:   because the following virtual functions are pure within 'soundtouch::InterpolateShannon':
RateTransposer.h:64: note: 	virtual int soundtouch::TransposerBase::transposeMono(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:67: note: 	virtual int soundtouch::TransposerBase::transposeStereo(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:70: note: 	virtual int soundtouch::TransposerBase::transposeMulti(soundtouch::SAMPLETYPE*, const soundtouch::SAMPLETYPE*, int&)
RateTransposer.h:86: note: 	virtual void soundtouch::TransposerBase::resetRegisters()
make[3]: *** [RateTransposer.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from ../../include/FIFOSampleBuffer.h:42,
                 from AAFilter.h:41,
                 from AAFilter.cpp:40:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
In file included from AAFilter.h:41,
                 from AAFilter.cpp:40:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
make[3]: *** [AAFilter.lo] Error 1
In file included from ../../include/SoundTouch.h:68,
                 from SoundTouch.cpp:71:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
In file included from SoundTouch.cpp:71:
../../include/SoundTouch.h:212: error: expected ';' before 'override'
In file included from ../../include/FIFOSampleBuffer.h:42,
                 from FIFOSampleBuffer.cpp:44:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'../../include/SoundTouch.h:290: error: expected ';' before 'override'

../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/SoundTouch.h:299: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
../../include/SoundTouch.h:307: error: expected ';' before 'override'
../../include/SoundTouch.h:311: error: expected ';' before 'override'
In file included from AAFilter.h:41,
                 from RateTransposer.h:42,
                 from TDStretch.h:42,
                 from SoundTouch.cpp:72:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
In file included from FIFOSampleBuffer.cpp:44:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
In file included from TDStretch.h:42,
                 from SoundTouch.cpp:72:
RateTransposer.h:127: error: expected ';' before 'override'
RateTransposer.h:150: error: expected ';' before 'override'
RateTransposer.h:153: error: expected ';' before 'override'
RateTransposer.h:156: error: expected ';' before 'override'
RateTransposer.h:107: error: cannot declare field 'soundtouch::RateTransposer::inputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   because the following virtual functions are pure within 'soundtouch::FIFOSampleBuffer':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
RateTransposer.h:110: error: cannot declare field 'soundtouch::RateTransposer::midBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
RateTransposer.h:113: error: cannot declare field 'soundtouch::RateTransposer::outputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
FIFOSampleBuffer.cpp:63: error: definition of implicitly-declared 'virtual soundtouch::FIFOSampleBuffer::~FIFOSampleBuffer()'
In file included from SoundTouch.cpp:72:
TDStretch.h:168: error: expected ';' before 'override'
TDStretch.h:190: error: expected ';' before 'override'
FIFOSampleBuffer.cpp:99: error: prototype for 'void soundtouch::FIFOSampleBuffer::putSamples(const soundtouch::SAMPLETYPE*, uint)' does not match any in class 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:133: error: candidate is: virtual void soundtouch::FIFOSampleBuffer::putSamples(uint)
TDStretch.h:230: error: expected ';' before 'override'
TDStretch.h:135: error: cannot declare field 'soundtouch::TDStretch::outputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
TDStretch.h:136: error: cannot declare field 'soundtouch::TDStretch::inputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
FIFOSampleBuffer.cpp:148: error: no 'soundtouch::SAMPLETYPE* soundtouch::FIFOSampleBuffer::ptrBegin()' member function declared in class 'soundtouch::FIFOSampleBuffer'
SoundTouch.cpp: In constructor 'soundtouch::SoundTouch::SoundTouch()':
SoundTouch.cpp:93: error: cannot allocate an object of abstract type 'soundtouch::RateTransposer'
RateTransposer.h:99: note:   because the following virtual functions are pure within 'soundtouch::RateTransposer':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
SoundTouch.cpp: At global scope:
SoundTouch.cpp:114: error: definition of implicitly-declared 'virtual soundtouch::SoundTouch::~SoundTouch()'
FIFOSampleBuffer.cpp:200: error: no 'uint soundtouch::FIFOSampleBuffer::numSamples() const' member function declared in class 'soundtouch::FIFOSampleBuffer'
FIFOSampleBuffer.cpp:211: error: no 'uint soundtouch::FIFOSampleBuffer::receiveSamples(soundtouch::SAMPLETYPE*, uint)' member function declared in class 'soundtouch::FIFOSampleBuffer'
FIFOSampleBuffer.cpp: In member function 'uint soundtouch::FIFOSampleBuffer::receiveSamples(soundtouch::SAMPLETYPE*, uint)':
FIFOSampleBuffer.cpp:218: error: no matching function for call to 'soundtouch::FIFOSampleBuffer::receiveSamples(uint&)'
FIFOSampleBuffer.cpp:211: note: candidates are: uint soundtouch::FIFOSampleBuffer::receiveSamples(soundtouch::SAMPLETYPE*, uint)
FIFOSampleBuffer.cpp: At global scope:
FIFOSampleBuffer.cpp:225: error: prototype for 'uint soundtouch::FIFOSampleBuffer::receiveSamples(uint)' does not match any in class 'soundtouch::FIFOSampleBuffer'
FIFOSampleBuffer.cpp:211: error: candidate is: uint soundtouch::FIFOSampleBuffer::receiveSamples(soundtouch::SAMPLETYPE*, uint)
FIFOSampleBuffer.cpp:244: error: no 'int soundtouch::FIFOSampleBuffer::isEmpty() const' member function declared in class 'soundtouch::FIFOSampleBuffer'
FIFOSampleBuffer.cpp:251: error: no 'void soundtouch::FIFOSampleBuffer::clear()' member function declared in class 'soundtouch::FIFOSampleBuffer'
FIFOSampleBuffer.cpp:260: error: no 'uint soundtouch::FIFOSampleBuffer::adjustAmountOfSamples(uint)' member function declared in class 'soundtouch::FIFOSampleBuffer'
mv -f .deps/cpu_detect_x86.Tpo .deps/cpu_detect_x86.Plo
SoundTouch.cpp:277: error: no 'void soundtouch::SoundTouch::putSamples(const soundtouch::SAMPLETYPE*, uint)' member function declared in class 'soundtouch::SoundTouch'
In file included from ../../include/FIFOSampleBuffer.h:42,
                 from BPMDetect.cpp:57:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
In file included from BPMDetect.cpp:57:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
SoundTouch.cpp:481: error: no 'void soundtouch::SoundTouch::clear()' member function declared in class 'soundtouch::SoundTouch'
SoundTouch.cpp:511: error: no 'uint soundtouch::SoundTouch::receiveSamples(soundtouch::SAMPLETYPE*, uint)' member function declared in class 'soundtouch::SoundTouch'
SoundTouch.cpp:524: error: prototype for 'uint soundtouch::SoundTouch::receiveSamples(uint)' does not match any in class 'soundtouch::SoundTouch'
SoundTouch.cpp:511: error: candidate is: uint soundtouch::SoundTouch::receiveSamples(soundtouch::SAMPLETYPE*, uint)
In file included from ../../include/FIFOSampleBuffer.h:42,
                 from AAFilter.h:41,
                 from RateTransposer.h:42,
                 from TDStretch.h:42,
                 from TDStretch.cpp:51:
../../include/FIFOSamplePipe.h:167: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:178: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:178: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:183: error: expected `;' before 'public'
../../include/FIFOSamplePipe.h:192: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:202: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:203: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:209: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:209: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:215: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:215: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:222: error: expected `;' before 'virtual'
../../include/FIFOSamplePipe.h:222: error: expected ';' before 'override'
../../include/FIFOSamplePipe.h:226: error: expected `;' before '}' token
In file included from AAFilter.h:41,
                 from RateTransposer.h:42,
                 from TDStretch.h:42,
                 from TDStretch.cpp:51:
../../include/FIFOSampleBuffer.h:94: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:103: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:124: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:142: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:150: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:153: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:165: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:168: error: expected ';' before 'override'
../../include/FIFOSampleBuffer.h:172: error: expected ';' before 'override'
In file included from TDStretch.h:42,
                 from TDStretch.cpp:51:
RateTransposer.h:127: error: expected ';' before 'override'
RateTransposer.h:150: error: expected ';' before 'override'
RateTransposer.h:153: error: expected ';' before 'override'
RateTransposer.h:156: error: expected ';' before 'override'
RateTransposer.h:107: error: cannot declare field 'soundtouch::RateTransposer::inputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   because the following virtual functions are pure within 'soundtouch::FIFOSampleBuffer':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
RateTransposer.h:110: error: cannot declare field 'soundtouch::RateTransposer::midBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
RateTransposer.h:113: error: cannot declare field 'soundtouch::RateTransposer::outputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
In file included from TDStretch.cpp:51:
TDStretch.h:168: error: expected ';' before 'override'
TDStretch.h:190: error: expected ';' before 'override'
TDStretch.h:230: error: expected ';' before 'override'
TDStretch.h:135: error: cannot declare field 'soundtouch::TDStretch::outputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
TDStretch.h:136: error: cannot declare field 'soundtouch::TDStretch::inputBuffer' to be of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   since type 'soundtouch::FIFOSampleBuffer' has pure virtual functions
TDStretch.cpp: In constructor 'soundtouch::TDStretch::TDStretch()':
TDStretch.cpp:99: warning: abstract virtual 'virtual void soundtouch::FIFOSamplePipe::clear()' called from constructor
TDStretch.cpp: At global scope:
TDStretch.cpp:104: error: definition of implicitly-declared 'virtual soundtouch::TDStretch::~TDStretch()'
TDStretch.cpp:228: error: no 'void soundtouch::TDStretch::clear()' member function declared in class 'soundtouch::TDStretch'
make[3]: *** [FIFOSampleBuffer.lo] Error 1
TDStretch.cpp: In member function 'void soundtouch::TDStretch::processSamples()':
TDStretch.cpp:706: error: no matching function for call to 'soundtouch::FIFOSampleBuffer::putSamples(soundtouch::SAMPLETYPE*, uint)'
../../include/FIFOSampleBuffer.h:133: note: candidates are: virtual void soundtouch::FIFOSampleBuffer::putSamples(uint)
TDStretch.cpp: At global scope:
TDStretch.cpp:728: error: no 'void soundtouch::TDStretch::putSamples(const soundtouch::SAMPLETYPE*, uint)' member function declared in class 'soundtouch::TDStretch'
TDStretch.cpp: In member function 'void soundtouch::TDStretch::putSamples(const soundtouch::SAMPLETYPE*, uint)':
TDStretch.cpp:731: error: no matching function for call to 'soundtouch::FIFOSampleBuffer::putSamples(const soundtouch::SAMPLETYPE*&, uint&)'
../../include/FIFOSampleBuffer.h:133: note: candidates are: virtual void soundtouch::FIFOSampleBuffer::putSamples(uint)
TDStretch.cpp: In static member function 'static soundtouch::TDStretch* soundtouch::TDStretch::newInstance()':
TDStretch.cpp:799: error: cannot allocate an object of abstract type 'soundtouch::TDStretch'
TDStretch.h:104: note:   because the following virtual functions are pure within 'soundtouch::TDStretch':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
make[3]: *** [SoundTouch.lo] Error 1
BPMDetect.cpp: In constructor 'soundtouch::BPMDetect::BPMDetect(int, int)':
BPMDetect.cpp:213: error: cannot allocate an object of abstract type 'soundtouch::FIFOSampleBuffer'
../../include/FIFOSampleBuffer.h:53: note:   because the following virtual functions are pure within 'soundtouch::FIFOSampleBuffer':
../../include/FIFOSamplePipe.h:78: note: 	virtual soundtouch::SAMPLETYPE* soundtouch::FIFOSamplePipe::ptrBegin()
../../include/FIFOSamplePipe.h:84: note: 	virtual void soundtouch::FIFOSamplePipe::putSamples(const soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:104: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(soundtouch::SAMPLETYPE*, uint)
../../include/FIFOSamplePipe.h:112: note: 	virtual uint soundtouch::FIFOSamplePipe::receiveSamples(uint)
../../include/FIFOSamplePipe.h:115: note: 	virtual uint soundtouch::FIFOSamplePipe::numSamples() const
../../include/FIFOSamplePipe.h:118: note: 	virtual int soundtouch::FIFOSamplePipe::isEmpty() const
../../include/FIFOSamplePipe.h:121: note: 	virtual void soundtouch::FIFOSamplePipe::clear()
../../include/FIFOSamplePipe.h:125: note: 	virtual uint soundtouch::FIFOSamplePipe::adjustAmountOfSamples(uint)
BPMDetect.cpp: In member function 'void soundtouch::BPMDetect::inputSamples(const soundtouch::SAMPLETYPE*, int)':
BPMDetect.cpp:431: error: no matching function for call to 'soundtouch::FIFOSampleBuffer::putSamples(soundtouch::SAMPLETYPE [256], int&)'
../../include/FIFOSampleBuffer.h:133: note: candidates are: virtual void soundtouch::FIFOSampleBuffer::putSamples(uint)
make[3]: *** [TDStretch.lo] Error 1
make[3]: *** [BPMDetect.lo] Error 1
mv -f .deps/FIRFilter.Tpo .deps/FIRFilter.Plo
make[3]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/source/SoundTouch'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2/source'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2'
make: *** [all] Error 2
make: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2'
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_audio_soundtouch/soundtouch/work/2.3.2" && /usr/bin/make -j8 -w all 
Exit code: 2
```

Reply:
@barracuda156 this is because you are trying to block GCC version `34` (which does not yet exist) instead of `3-4` 😄

Reply:
@aeiouaeiouaeiouaeiouaeiouaeiou No-no. Firstly, `*gcc-[34].*` is correct and works as intended. The log above is a result of removing blacklist in favor of dropping `-Ofast` flag (locally on my machine). My point is that without the flag old GCC still fails. With blacklist – as it is in the PR – everything works fine.

Reply:
@aeiouaeiouaeiouaeiouaeiouaeiou @catap I mean, we can still drop the flag, no objections here, it is just unsufficient.